packages/ic-pub-key/src/lib.rs

```diff
@@ -0,0 +1,415 @@
+#![forbid(unsafe_code)]
+#![forbid(missing_docs)]
+#![warn(clippy::all)]
+#![warn(clippy::nursery)]
+#![warn(future_incompatible)]
+#![warn(rust_2018_idioms)]
+#![warn(rustdoc::broken_intra_doc_links)]
+#![warn(rustdoc::missing_crate_level_docs)]
+#![deny(unused_must_use)]
+#![deny(unused_results)]
+
+//! A crate for performing derivation of threshold public keys
+
+#[cfg(not(any(feature = "secp256k1", feature = "ed25519", feature = "vetkeys")))]
+compile_error!("At least one of the features (secp256k1, ed25519, vetkeys) must be enabled");
+
+pub use ic_management_canister_types::{
+    CanisterId, EcdsaCurve, EcdsaKeyId, EcdsaPublicKeyArgs, EcdsaPublicKeyResult, SchnorrAlgorithm,
+    SchnorrKeyId, SchnorrPublicKeyArgs, SchnorrPublicKeyResult, VetKDCurve, VetKDKeyId,
+    VetKDPublicKeyArgs, VetKDPublicKeyResult,
+};
+
+/// Error that can occur during public key derivation
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub enum Error {
+    /// The specified master public key is not known
+    UnknownKeyIdentifier,
+    /// The algorithm is not supported (possibly due to an unset feature)
+    AlgorithmNotSupported,
+    /// Currently the canister must be specified
+    CanisterIdMissing,
```

Comment:
We cannot make it so that the canister ID is actually optional, can we? So, I'd remove the word "Currently" and maybe add a short explanation of why the canister ID must be specified, which in my understanding is because the library doesn't know the ID of the calling canister (in contrast to the management canister methods *_public_key, who do know the ID of the calling canister).

Reply:
Yeah I guess there really isn't a practical scenario where we will be able to deduce the canister id in this library context

Reply:
Updated/expanded the comment